### PR TITLE
Add tests for VPC Endpoint Service Configuration cleanup

### PR DESCRIPTION
**Pull request by NautiluX**
jira: [OSD-10055](https://issues.redhat.com/browse/OSD-10055)

### CODE REVIEW

File: pkg/controller/accountclaim/reuse_test.go [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+				describeOutput = ec2.DescribeVpcEndpointServiceConfigurationsOutput{
+					ServiceConfigurations: []*ec2.ServiceConfiguration{
+						{
+							ServiceId: &serviceConfigId,
+						},
+					},
+				}
+				deleteOutput = ec2.DeleteVpcEndpointServiceConfigurationsOutput{}
+				mockAwsClient.EXPECT().DescribeVpcEndpointServiceConfigurations(gomock.Any()).Return(&describeOutput, nil)
+				mockAwsClient.EXPECT().DeleteVpcEndpointServiceConfigurations(gomock.Any()).Do(func(input *ec2.DeleteVpcEndpointServiceConfigurationsInput) {
```
Comment (dkeohane) [inline pkg/controller/accountclaim/reuse_test.go:101]: Ah nice, I hadn't seen the `.Do` with the inner function used before to save the input - I've always gone around it with using specific Expect calls for the input, but this is nice to push the Expect calls to after.

File: pkg/controller/accountclaim/reuse_test.go [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+				notifications, errors, _ := runCleanupFunc(r.CleanUpAwsAccountVpcEndpointServiceConfigurations, mockAwsClient)
+
+				Expect(len(deleteInput.ServiceIds)).To(Equal(1))
+				Expect(*deleteInput.ServiceIds[0]).To(Equal(serviceConfigId))
+				Expect(errors).To(Equal(""))
+				Expect(notifications).To(Equal("VPC endpoint service configuration cleanup finished successfully"))
+			})
+		})
+
+		Context("When two VPC Endpoint Service Configuration exist", func() {
```
Comment (dkeohane) [inline pkg/controller/accountclaim/reuse_test.go:116]: Just missing 1 other test where we cover if one of the VPC enpoint deletions returns an error.

**Review by dkeohane — COMMENTED** [review]
Thanks for adding in the tests - I probably jumped the gun by approving the previous PR before you had them added!

File: pkg/controller/accountclaim/reuse_test.go (hunk 2 above)
Comment (NautiluX) [reply pkg/controller/accountclaim/reuse_test.go:118]: done

**Review by dkeohane — APPROVED** [review]
/lgtm

**Pull request merged**